### PR TITLE
OpenZFS 4521 - zfstest is trying to execute evil "zfs unmount -a"

### PR DESCRIPTION
**Pull request by dinatale2**
Authored by: Yuri Pankov <yuri.pankov@nexenta.com>
Reviewed by: Andriy Gapon <avg@FreeBSD.org>
Reviewed by: Dan McDonald <danmcd@omniti.com>
Reviewed by: Matthew Ahrens <mahrens@delphix.com>
Reviewed by: John Kennedy <john.kennedy@delphix.com>
Approved by: Robert Mustacchi <rm@joyent.com>
Ported-by: Giuseppe Di Natale <dinatale2@llnl.gov>

OpenZFS-issue: https://www.illumos.org/issues/4521
OpenZFS-commit: https://github.com/openzfs/openzfs/commit/8808ac5

### CODE REVIEW

**Comment by behlendorf** [comment]
This patch causes `inherit_001_pos` to fail on all builders.  I haven't looked that closely but it may be related to the `zfstest.ksh` changes which were dropped when the patch was ported.  Similar functional changes need to be made the the Linux equivalent of this script `scripts/zfs-tests.sh`.

```
Test: /usr/share/zfs/zfs-tests/tests/functional/inheritance/inherit_001_pos (run as root) [00:14] [FAIL]
```

You'll also want to rebase this on master.

File: tests/zfs-tests/tests/functional/inheritance/inherit_001_pos.ksh [hunk 1]
```diff
@@ -331,7 +331,7 @@ function scan_state { #state-file
 				if [[ $op == "-" ]]; then
 					log_note "No operation specified"
 				else
-					export __ZFS_POOL_RESTRICT="$TESTPOOL"
+					export __ZFS_POOL_RESTRICT="TESTPOOL"
```
Comment (dinatale2) [inline tests/zfs-tests/tests/functional/inheritance/inherit_001_pos.ksh]: @jwk404 Does inherit_001_pos work for OpenZFS? I noticed that this is the only place "TESTPOOL" was being expanded.  It appears that else where in the script, the pool name is expected to be "TESTPOOL".

Comment (jwk404) [inline tests/zfs-tests/tests/functional/inheritance/inherit_001_pos.ksh]: At Delphix, we have the change you describe above (using "TESTPOOL") and the test passes. We haven't yet upstreamed the change that introduced the fix to OpenZFS yet though.

Comment (behlendorf) [inline tests/zfs-tests/tests/functional/inheritance/inherit_001_pos.ksh]: Thanks for the confirmation!

**Pull request merged**